File: back_end/src/main/java/com/fstg/restaurant/model/Order.java
package com.fstg.restaurant.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "orders")
public class Order extends CategoryOrder {

	
	
	@Column(name = "price")
	private int price;
	
	@Column(name = "image")
	private String img;
	
	@Column(name = "description")
	@Lob
	private String description;
	
	@JsonIgnore//pour ignorer ce champ lors du reponse pour eviter une boucle infinie de ce champ
	@ManyToOne
	@JoinColumn(name = "id_category")
	private Category category;
	
}

File: front_end/src/app/model/order.ts
export class Order {
  id: number;
  data_create: Date;
  data_update: Date;
  name: string;
  description: string;
  img: string;
  price: number;
  // id_category
  constructor(
    id_: number,
    data_create_: Date,
    data_update_: Date,
    name_: string,
    description_: string,
    image: string,
    price_: number
  ) {
    this.id = id_;
    this.data_create = data_create_;
    this.data_update = data_update_;
    this.name = name_;
    this.description = description_;
    this.img = image;
    this.price = price_;
  }
}
